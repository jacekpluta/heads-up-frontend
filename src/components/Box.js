import React, { useState } from "react";
import GameModule from "./GameModule";

import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from "react-router-dom";

function Box(props) {
  const [clickedTrue, setClickedTrue] = useState(false);

  const handleClickBox = () => {
    setClickedTrue(true);
  };

  return (
    <div className="Box" onClick={handleClickBox}>
      <Router>
        <Switch>
          <Route exact path="/GameModule">
            <GameModule></GameModule>
          </Route>
          {clickedTrue ? <Redirect to="/GameModule" /> : <Redirect to="/" />}
        </Switch>
      </Router>
    </div>
  );
}

//{this.state.showComponent ? <GameModule /> : null}
export default Box;
