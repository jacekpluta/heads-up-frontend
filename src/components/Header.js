import React, { Component } from "react";

class Header extends Component {
  state = {};
  render() {
    return <div className="Header"></div>;
  }
}

export default Header;
