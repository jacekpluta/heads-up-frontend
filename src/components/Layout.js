import styled from "styled-components";

export const DivLink = styled.div`
  cursor: pointer;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  text-decoration: none;
  z-index: 1;
  background-color: white;
  opacity: 0;
  filter: alpha(opacity=0);
`;
