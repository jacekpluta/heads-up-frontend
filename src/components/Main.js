import React, { Component } from "react";
import Box from "./Box";
import gamePicTwo from "../pic/gamePicTwo.jpg";

class Main extends Component {
  render() {
    const numbers = [1];

    return (
      <div className="Main">
        {numbers.map(key => {
          return (
            <Box
              key={key}
              style={{ backgroundImage: `url(${gamePicTwo})` }}
            ></Box>
          );
        })}
      </div>
    );
  }
}

export default Main;
