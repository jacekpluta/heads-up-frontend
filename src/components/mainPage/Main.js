import React from "react";
import Box from "./Box";

const Main = (props) => {
  const { gameCategoriesList } = props;

  return (
    <div className="main">
      {gameCategoriesList.map((gameCategory) => (
        <div className="mainItem">
          <Box
            backgroundImage={gameCategory.background}
            gameCategory={gameCategory}
          ></Box>
        </div>
      ))}
    </div>
  );
};

export default Main;
