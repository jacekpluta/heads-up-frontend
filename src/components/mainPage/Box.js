import React, { useState, useEffect, useContext } from "react";
import { motion } from "framer-motion";
import UIfx from "uifx";
import buttonClick from "../../sounds/buttonClick.mp3";
import { DivLink } from "../../styles/Layout";
import { GameCategoryContext } from "../../contex/GameCategoryContext";

import { useHistory } from "react-router-dom";

function Box(props) {
  const [gameCategoryPicked, setGameCategoryPicked] = useState(false);

  const { setGameCategory } = useContext(GameCategoryContext);

  const { backgroundImage, gameCategory } = props;

  let history = useHistory();

  //SOUDS
  const clickSound = new UIfx(buttonClick, {
    volume: 1,
    throttleMs: 100,
  });

  //plays click on box click
  const boxClicked = () => {
    if (gameCategory) {
      clickSound.play();
    }
  };

  //OPEN GAME MENU AFTER PICKING GAME CATEGORY
  const handleGameCategoryPicked = () => {
    setGameCategoryPicked(!gameCategoryPicked);
    setGameCategory(gameCategory);
  };

  useEffect(() => {
    if (gameCategoryPicked) {
      history.push("/gamemenu");
    }
  }, [gameCategoryPicked, history]);

  return (
    <motion.div
      style={backgroundImage}
      className="box"
      whileHover={{ scale: 1.05 }}
      onClick={() => boxClicked()}
    >
      <DivLink onClick={handleGameCategoryPicked}> </DivLink>
    </motion.div>
  );
}

//{this.state.showComponent ? <GameModule /> : null}
export default Box;
