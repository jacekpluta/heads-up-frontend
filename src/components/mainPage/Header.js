import React from "react";
// import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
// import {
//   faVolumeUp,
//   faQuestionCircle,
// } from "@fortawesome/free-solid-svg-icons";
import { motion } from "framer-motion";
import { HeaderStyle } from "../../styles/Layout";
// import { MuteSoundContext } from "../../contex/MuteSoundContext";

// import { clickSound } from "../Sounds";
// import HowToPlay from "./HowToPlay";
import HowToPlayPic from "../../pic/HowToPlayPic.jpg";
import { Button, Popup, Grid } from "semantic-ui-react";

function Header(props) {
  // const { muteSound, setMuteSound } = useContext(MuteSoundContext);
  // const [showHowToPlay, setShowHowToPlay] = useState(false);
  // const handleMuteSound = () => {
  //   if (!muteSound) {
  //     clickSound.play();
  //   }
  //   setMuteSound(!muteSound);
  // };

  // const volumeButtonStyle = {
  //   float: "top-left",
  //   marginLeft: "2%",
  //   marginTop: "33px",
  //   backgroundColor: "#1b85ff",
  //   borderColor: " #1b63ff",
  //   borderStyle: "ridge",
  //   borderRadius: "15px",
  //   borderWidth: "4px",
  //   padding: "3px",
  // };

  const containerStyle = {
    textAlign: "center",
    position: "absolute",
    right: 0,
    top: "40px",
    bottom: 0,
  };
  const contextRef = React.useRef();
  return (
    <HeaderStyle>
      <Popup
        trigger={
          <motion.img
            whileHover={{ scale: 1.2 }}
            whileTap={{
              scale: 0.95,
            }}
            src={HowToPlayPic}
            alt="How To Play"
            width="200"
            height="200"
            style={containerStyle}
          />
        }
        position="top right"
        flowing
        hoverable
        wide
        context={contextRef}
      >
        <Grid centered divided columns={3}>
          <Grid.Column textAlign="center">
            <p>
              <Button color="blue" size="mini">
                1
              </Button>
            </p>
            <p>Pick game category and game variant</p>
          </Grid.Column>

          <Grid.Column textAlign="center">
            <p>
              <Button color="blue" size="mini">
                2
              </Button>
            </p>
            <p>Place your phone on your forehead</p>
          </Grid.Column>

          <Grid.Column textAlign="center">
            <p>
              <Button color="blue" size="mini">
                3
              </Button>
            </p>
            <p>
              Did u guess codeword? Tilt your phone forward or click "CORRECT"
              button
            </p>
            <p>Do you want to skip current codeword? Click "SKIP" button</p>
          </Grid.Column>
        </Grid>
      </Popup>
      <strong ref={contextRef} style={containerStyle}></strong>
    </HeaderStyle>
  );
}

export default Header;
