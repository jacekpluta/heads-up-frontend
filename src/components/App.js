import React, { useEffect, useState } from "react";
import "../styles/styles.css";
import { CircularProgress, Grid } from "@material-ui/core/";

import Header from "./Header";
import Main from "./Main";

import AnimalsTile from "../pic/animalsTile.jpg";
import MoviesTile from "../pic/moviesTile.jpg";
import GamesTile from "../pic/gamesTile.jpg";
import AnimeTile from "../pic/animeTile.jpg";

import { ParStyle, BlueBackgroundStyle } from "../styles/Layout";
import { isBrowser } from "react-device-detect";
import animalList from "./lists/AnimalsList";
import axios from "axios";
import { motion } from "framer-motion";

function App() {
  const [filmList, setFilmList] = useState([]);
  const [gameList, setGameList] = useState([]);
  const [animeList, setAnimeList] = useState([]);

  const [animeFetched, setAnimeFetched] = useState(false);
  const [filmsFetched, setFilmsFetched] = useState(false);
  const [gamesFetched, setGamesFetched] = useState(false);
  const [allFechted, setAllFechted] = useState(false);
  const [muteSounds, setMuteSounds] = useState(false);

  const gameVariantsList = [
    {
      id: 0,
      name: "animals",
      gameMenuTitle: "Zwierzeta",
      questions: animalList,
      background: { backgroundImage: `url(${AnimalsTile})` },
      gameTile: { AnimalsTile },
      description: "Czy umiesz udawać słonia? ",
    },
    {
      id: 1,
      name: "films",
      gameMenuTitle: "Filmy",
      questions: filmList,
      background: { backgroundImage: `url(${MoviesTile})` },
      gameTile: { MoviesTile },
      description: "aaa",
    },
    {
      id: 2,
      name: "games",
      gameMenuTitle: "Gry",
      questions: gameList,
      background: { backgroundImage: `url(${GamesTile})` },
      gameTile: { GamesTile },
      description: "aaa",
    },
    {
      id: 3,
      name: "anime",
      gameMenuTitle: "Anime",
      questions: animeList,
      background: { backgroundImage: `url(${AnimeTile})` },
      gameTile: { AnimeTile },
      description: "aaa",
    },
  ];

  async function fetchMyAPIFilms() {
    axios({
      method: "GET",
      url:
        "https://api.themoviedb.org/3/movie/popular?api_key=01e57b363d38e654e9afbd273dce30c3&language=en-US&page=1",
    })
      .then((response) => {
        Object.entries(response.data.results).map(([key, value]) =>
          setFilmList((filmList) => [
            ...filmList,
            response.data.results[key].name,
          ])
        );
      })
      .then(() => {
        setFilmsFetched(true);
      })
      .catch((error) => {
        console.log(error);
      });
  }

  async function fetchMyAPIGames() {
    axios({
      method: "GET",
      url: "https://rawg-video-games-database.p.rapidapi.com/games",
      headers: {
        "content-type": "application/octet-stream",
        "x-rapidapi-host": "rawg-video-games-database.p.rapidapi.com",
        "x-rapidapi-key": "e7ded30215msh9c35219cd4db934p1c9604jsn6de6432b8ee4",
      },
    })
      .then((response) => {
        Object.entries(response.data.results).map(([key, value]) =>
          setGameList((gameList) => [
            ...gameList,
            response.data.results[key].name,
          ])
        );
      })
      .then(() => {
        setGamesFetched(true);
      })
      .catch((error) => {
        console.log(error);
      });
  }

  async function fetchMyAPIAnime() {
    axios({
      method: "GET",
      url: "https://kitsu.io/api/edge/anime?page[limit]=20&page[offset]=0",
      headers: {
        Accept: "application/vnd.api+json",
        "Content-type": "application/vnd.api+json",
        links: {
          first:
            "https://kitsu.io/api/edge/anime?page[limit]=20&page[offset]=1",
        },
      },
    })
      .then((response) => {
        Object.entries(response.data.data).map(([key, value]) =>
          setAnimeList((animeList) => [
            ...animeList,
            response.data.data[key].attributes.titles.en_jp,
          ])
        );
      })
      .then(() => {
        setAnimeFetched(true);
      })
      .catch((error) => {
        console.log(error);
      });
  }
  const handleMuteSounds = () => {
    setMuteSounds(!muteSounds);
  };

  useEffect(() => {
    if (!isBrowser) {
      window.screen.orientation.lock("landscape");
      // fetchMyAPIFilms();
      // fetchMyAPIAnime();
      // fetchMyAPIGames();
    }
  }, []);

  useEffect(() => {
    if (animeFetched && gamesFetched && filmsFetched) {
      setAllFechted(true);
    }
  }, [animeFetched, gamesFetched, filmsFetched]);

  const pageTransition = {
    inModule: {
      opacity: 1,
      transition: {
        duration: 0.3,
      },
    },
    outModule: {
      opacity: 0,
      transition: {
        duration: 0.3,
      },
    },
  };
  if (isBrowser) {
    return (
      <BlueBackgroundStyle>
        <ParStyle style={{ fontSize: "5vw" }}>
          Gra działa tylko na urządzeniach mobilnych
        </ParStyle>
      </BlueBackgroundStyle>
    );
  } else if (!allFechted) {
    return (
      <motion.div
        className="App"
        variants={pageTransition}
        initial={"outModule"}
        animate={"inModule"}
        exit={"outModule"}
      >
        <Header handleMuteSounds={handleMuteSounds} />

        <Main
          allFechted={allFechted}
          gameVariantsList={gameVariantsList}
          muteSounds={muteSounds}
        ></Main>
      </motion.div>
    );
  } else
    return (
      <Grid
        container
        spacing={0}
        alignItems="center"
        justify="center"
        style={{
          background: "linearGradient(180deg, #013064, #1255a0)",
          minHeight: "100vh",
        }}
      >
        <CircularProgress thickness={5} color="secondary" />
      </Grid>
    );
}

export default App;
