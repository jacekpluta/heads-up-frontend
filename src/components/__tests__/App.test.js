import React, { Component } from "react";
import TestRenderer from "react-test-renderer";

import App from "../App";
import Header from "../mainPage/Header";
import Main from "../mainPage/Main";

import Box from "../mainPage/Box";
import { HeaderStyle } from "../../styles/Layout";

import { MemoryRouter } from "react-router-dom";
import * as deviceDetect from "react-device-detect";
import { GameCategoryContext } from "../../contex/GameCategoryContext";
import { MuteSoundContext } from "../../contex/MuteSoundContext";

import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

deviceDetect.isMobile = true;

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
});

afterEach(() => {});

// it("should render itself", () => {
//   const gameCategoryValue = {
//     id: 0,
//     name: "animals",
//     gameMenuTitle: "Zwierzeta",
//     questions: [],
//     background: "",
//     gameTile: "",
//     description: "Czy umiesz udawać słonia?",
//   };
// });

it("render correctly text component", () => {
  const gameCategoryValue = {
    id: 0,
    name: "animals",
    gameMenuTitle: "Zwierzeta",
    questions: [],
    background: "",
    gameTile: "",
    description: "Czy umiesz udawać słonia?",
  };

  const TextInputComponent = TestRenderer.create(
    <MuteSoundContext.Provider value={false}>
      <GameCategoryContext.Provider value={gameCategoryValue}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </GameCategoryContext.Provider>
    </MuteSoundContext.Provider>
  ).toJSON();
  expect(TextInputComponent).toMatchSnapshot();
});

//   <GameCategoryContext.Provider value={gameCategoryValue}>
//
// <MuteSoundContext.Provider value={false}>
