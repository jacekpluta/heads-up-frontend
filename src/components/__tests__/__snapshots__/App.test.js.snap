// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly text component 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  style={
    Object {
      "background": "linearGradient(180deg, #013064, #1255a0)",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style={
      Object {
        "background": "linearGradient(180deg, #013064, #1255a0)",
      }
    }
  >
    <p
      className="sc-bwzfXH fKrVns"
      style={
        Object {
          "fontSize": "5vw",
          "top": "40%",
        }
      }
    >
      Loading the game
    </p>
    <p
      className="sc-bwzfXH fKrVns"
      style={
        Object {
          "fontSize": "5vw",
          "top": "60%",
        }
      }
    >
      
    </p>
    <div
      style={
        Object {
          "color": "#fff",
          "fontSize": "5vw",
          "left": "50%",
          "position": "absolute",
          "textAlign": "center",
          "textShadow": "3px 3px 0px rgba(0, 0, 0, 0.2)",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
        }
      }
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
